File: 2023/day_1/1.py
file = open("input1.txt").read()
lines = file.splitlines()
tempstr = ""
finalnum = 0
for line in lines:
    for character in line:
        try:
            int(character)
            tempstr += character
        except:
            continue
    firstnum = tempstr[0]
    lastnum = tempstr[(len(tempstr) - 1)]
    finalnum += int(firstnum+lastnum)
    print(f"String: {line}, tempstr: {tempstr}, first: {firstnum}, last: {lastnum}, num: {firstnum+lastnum}")
    tempstr = ""

print(finalnum)

File: 2023/day_1/1.rs
use std::fs;
use std::string;
fn main() {
    let mut tempstr = String::new();
    let mut lastnum: char;
    let mut firstnum: char;
    let mut finalnum = 0;
    let contents = fs::read_to_string("input1.txt")
        .expect("Should have been able to read file");
    let parts = contents.split("\n");
    for part in parts {
        for character in part.chars() {
            if character.is_numeric() {
                tempstr.push(character);
            };            
        };
        
        firstnum = tempstr.chars().nth(0);
        lastnum = tempstr.chars().nth((tempstr.len())-1);
    };
}
